### ISSUE TITLE
apollo-server-cache-redis: Add redis connection uri interface to RedisCache

### ISSUE THREAD

**Issue by airtoxin** [apollographql/apollo-server#5006]
`RedisCache` currently only accept `RedisOptions` interface of `ioredis`, but `Redis` constructor of `ioredis` can accept more variadic arguments.
https://github.com/luin/ioredis#connect-to-redis

```typescript
new (port?: number, host?: string, options?: IORedis.RedisOptions): IORedis.Redis;
new (host?: string, options?: IORedis.RedisOptions): IORedis.Redis;
new (options?: IORedis.RedisOptions): IORedis.Redis;
```

In my use case, my application already connect to redis via connection uri `redis://...`, and `RedisCache` also want to connect its redis, but `RedisCache` can't accept connection uri.

**Comment by glasser** [comment]
Hi! Rather than continuing down the road of making our package's API even more tightly coupled to the constructor API of `ioredis`, I'd rather see us expect the user to construct their own `ioredis` object and pass it to us. #4871 is a start at implementing this but as mentioned in my comment there the API is a bit awkward. I'm not sure that the original author on that PR is interested in continuing on it so somebody else might want to take a shot at it.

**Comment by glasser** [comment]
I've released a prerelease with this fix (`apollo-server-cache-redis@1.4.0-alpha.0`, part of the Apollo Server 2.23.0 release). Try out the alpha and see if it works for you! Please provide any feedback in #5094.

**Comment by glasser** [comment]
This has been released in `apollo-server-cache-redis@1.4.0`.
